SConstruct

```diff
@@ -0,0 +1,152 @@
+#!/usr/bin/env python
+import os
+import sys
+import subprocess
+
+# Global Setting.
+GODOT_CPP_REF = "godot-4.3-stable"
+PROJECT_DIR = "project"
+EXTENSION_NAME = "sentrysdk"
+COMPATIBILITY_MINIMUM = "4.3"
+
+BIN_DIR = "{project_dir}/addons/{extension_name}/bin".format(
+    project_dir=PROJECT_DIR,
+    extension_name=EXTENSION_NAME)
+
+# Checkout godot-cpp...
+if not os.path.exists("godot-cpp"):
+    print("Cloning godot-cpp repository...")
+    result = subprocess.run(
+        ["git", "clone", "-b", GODOT_CPP_REF, "https://github.com/godotengine/godot-cpp.git"],
+        check=True,
+    )
+    if result.returncode != 0:
+        print("Error: Cloning godot-cpp repository failed.")
+        Exit(1)
+    print("Finished cloning godot-cpp repository.")
+
+env = SConscript("godot-cpp/SConstruct")
+
+# Build sentry-native.
+# TODO: macOS needs to use a different SDK.
+if env["platform"] in ["linux", "macos"]:
+
+    def build_sentry_native(target, source, env):
+        result = subprocess.run(
+            ["sh", "scripts/build-sentry-native.sh"],
```

Comment:
Both can be used, really. I figured it's better to stick to something that is guaranteed to exist on a target platform. PowerShell is not really a standard on Linux, for instance, but shell is, and vice versa. On my distro, I don't even have a native powershell package - putting it as a requirement would create an unnecessary barrier. I could try rewriting it in Python - not sure how it would interact with Windows, but it's a possibility.